system/lib/compiler-rt/emscripten_setjmp.c

```diff
@@ -12,7 +12,8 @@
 static uint32_t setjmpId = 0;
 
 typedef struct TableEntry {
-  uint32_t id, label;
+  uintptr_t id;
```

Comment:
It looks like this is an auto-incrementing counter.. set from the `uint32_t setjmpId` above.   If so why does it need to change type?  At least I would expect `setjmpId` to be of the same type.

Reply:
It's tricky... It looks I need to turn even more variables into `uintptr_t`.

1. So I described in https://reviews.llvm.org/D101985 how changing the first argument of `emscripten_longjmp` in turn changed `testSetjmp`'s first argument to `uintptr_t`.
2. `testSetjmp`'s first argument `id` is compared with the local variable `curr`. So it's `uintptr_t` now: [Link](https://github.com/emscripten-core/emscripten/blob/10d4acf60e42ac88c955b01398f9d958fbca41b0/system/lib/compiler-rt/emscripten_setjmp.c#L51)
3. That `curr` is set from `table[i].id`, which is `TableEntry.id`. So `id` in `TableEntry` here is now `uintptr_t`: [Link](https://github.com/emscripten-core/emscripten/blob/10d4acf60e42ac88c955b01398f9d958fbca41b0/system/lib/compiler-rt/emscripten_setjmp.c#L49)
4. Because `table[i].id` is set from `setjmpId`, `setjmpId` needs to be `uintptr_t` too: [Link](https://github.com/emscripten-core/emscripten/blob/10d4acf60e42ac88c955b01398f9d958fbca41b0/system/lib/compiler-rt/emscripten_setjmp.c#L29)
5. Because `setjmpId` is stored in `env` (first parameter) in `saveSetjmp`, `saveSetjmp`'s first argument has to change to `uintptr_t`: [Link](https://github.com/emscripten-core/emscripten/blob/10d4acf60e42ac88c955b01398f9d958fbca41b0/system/lib/compiler-rt/emscripten_setjmp.c#L26)

5 was missing in this PR so I added it too. Not sure if this propagation of `uintptr_t` is good... WDYT?